src/ATen/native/xpu/Copy.cpp

```diff
@@ -294,13 +294,45 @@ void _copy_xpu(TensorIterator& iter, bool non_blocking) {
 }
 
 Tensor& _copy_xpu(Tensor& self, const Tensor& src, bool non_blocking) {
-  // TODO: valid check
+  TORCH_CHECK(self.defined(), "self is undefined");
+  TORCH_CHECK(src.defined(), "src is undefined");
+
+  // Copies into meta self are OK and just ignored (similar to inplace)
+  if (self.is_meta()) {
+    auto shape = infer_size_symdimvector(self.sym_sizes(), src.sym_sizes());
+    TORCH_CHECK(
+        self.sym_sizes().equals(shape),
+        "output with shape ",
+        self.sym_sizes(),
+        " doesn't match the broadcast shape ",
+        shape);
+    return self;
+  }
+
+  if (src.is_meta()) {
+    TORCH_CHECK_NOT_IMPLEMENTED(false, "Cannot copy out of meta tensor; no data!")
+  }
```

Comment:
We don't need it. Since, Meta dispatch key shares the code with CPU/CUDA. Here is XPU specific code path.